wdl-lint/src/rules/redundant_input_assignment.rs

```diff
@@ -0,0 +1,90 @@
+//! A lint rule for flagging redundant input assignments
+
+use std::fmt::Debug;
+use wdl_ast::{AstToken, VersionStatement};
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Span;
+use wdl_ast::SupportedVersion;
+use wdl_ast::v1::{Decl, InputSection};
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the Redundant Input Assignment rule.
+const ID: &str = "RedundantInputAssignment";
+
+/// Create a "Redundant Input Assignment" diagnostic.
+fn redundant_input_assignment(span: Span) -> Diagnostic {
+    Diagnostic::note("redundant input assignments can be shortened")
+        .with_rule(ID)
+        .with_label("redundant input assignments can be shortened", span)
+}
+
+/// Detects a malformed lint directive.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct RedundantInputAssignment {
+    included_version: bool,
+}
+
+impl Rule for RedundantInputAssignment {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Flags redundant input assignments."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Input assignments that are redundant can be shortened. For example, `{ input: a = a }` \
+        can be shortened to `{ input: a }`."
+    }
+
+    fn tags(&self) -> TagSet {
+        TagSet::new(&[Tag::Clarity, Tag::Correctness])
+    }
+
+    fn exceptable_nodes(&self) -> Option<&'static [wdl_ast::SyntaxKind]> {
+        None
+    }
+}
+
+impl Visitor for RedundantInputAssignment {
+    type State = Diagnostics;
+
+    fn document(&mut self, _: &mut Self::State, _: VisitReason, _: &Document, _: SupportedVersion) {
+        // This is intentionally empty, as this rule has no state.
+    }
+
+    fn version_statement(&mut self, state: &mut Self::State, reason: VisitReason, stmt: &VersionStatement) {
+        // this current takes into account for any possible patch versions, which might be unnecessary
+        let binding = stmt.version();
+        let split_versions = binding.as_str().split(".").collect::<Vec<_>>();
+        if split_versions[0] == "1" && split_versions[1] == "0" {
+            self.included_version = false;
+        } else {
+            self.included_version = true;
+        }
+    }
+
+    fn input_section(&mut self, state: &mut Self::State, reason: VisitReason, section: &InputSection) {
+        /// if the version statement is earlier than 1.1 return
+        if !self.included_version {
+            return
+        }
```

Comment:
@a-frantz is this the recommended way to access the state of the Visitor with regards to its version statement? Is there a guarantee that `version_statement` function is called before `input_section`?

Reply:
There's a better way to do the version check. You're going about it rather manual, but we've written an API layer for doing it that will be more robust and readable. Will link the docs in a moment.

> Is there a guarantee that version_statement function is called before input_section?

Yes there is! The version statement _must_ be the first non-trivia item in a document, so you can rely on the guarantee that the visitor will see the version statement before the visitor gets to an input section.

Reply:
You can use this API to do facilitate the version check for you: https://docs.rs/wdl/latest/wdl/ast/version/enum.V1.html

i.e. do not try to parse the string of the version, we've already written that logic with all the proper checks and what not.

Reply:
Check out how this rule uses the Version API:
https://github.com/stjude-rust-labs/wdl/blob/main/wdl-lint/src/rules/missing_requirements.rs#L75